### ISSUE TITLE
Elastic search mongo-connector error

### ISSUE THREAD

**Issue by TissonMathew** [10gen-labs/mongo-connector#211]
Any idea what I might be doing wrong here?

appserver232@apps:/$ mongo-connector -m mongodb://user:pass@server1:64697,server1:64696/catalogdb -t https://apikey@elastic-search-server.io -d elastic_doc_manager.py -n catalogdb.summary --auto-commit-interval=60

No handlers could be found for logger "mongo_connector.util"
Traceback (most recent call last):
  File "/usr/local/bin/mongo-connector", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/util.py", line 85, in wrapped
    func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/connector.py", line 950, in main
    conf.parse_args()
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/config.py", line 114, in parse_args
    option, dict((k, values.get(k)) for k in option.cli_names))
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/connector.py", line 778, in apply_doc_managers
    module = import_dm_by_name(dm['docManager'])
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/connector.py", line 768, in import_dm_by_name
    "Could not import %s." % full_name)
mongo_connector.errors.InvalidConfiguration: Could not import mongo_connector.doc_managers.elastic_doc_manager.py.

**Comment by TissonMathew** [comment]
Made few changes and now I'm getting the error below.

[ec2-user@ip-xxx-16-0-xxx ~]$ mongo-connector -m server1:64697 -t https://elastic-search-server -d elastic_doc_manager -n catalogdb.summary --admin-username user --password pass --auto-commit-interval=60
/usr/lib/python2.6/site-packages/elasticsearch/connection/http_urllib3.py:50: UserWarning: Connecting to elastic-search-server using SSL with verify_certs=False is insecure.
  'Connecting to %s using SSL with verify_certs=False is insecure.' % host)
No handlers could be found for logger "elasticsearch"
Traceback (most recent call last):
  File "/usr/bin/mongo-connector", line 11, in <module>
    sys.exit(main())
  File "/usr/lib/python2.6/site-packages/mongo_connector/util.py", line 85, in wrapped
    func(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/mongo_connector/connector.py", line 950, in main
    conf.parse_args()
  File "/usr/lib/python2.6/site-packages/mongo_connector/config.py", line 114, in parse_args
    option, dict((k, values.get(k)) for k in option.cli_names))
  File "/usr/lib/python2.6/site-packages/mongo_connector/connector.py", line 789, in apply_doc_managers
    dm_instances.append(module.DocManager(target_url, **kwargs))
  File "/usr/lib/python2.6/site-packages/mongo_connector/doc_managers/elastic_doc_manager.py", line 65, in __init__
    self.run_auto_commit()
  File "/usr/lib/python2.6/site-packages/mongo_connector/doc_managers/elastic_doc_manager.py", line 279, in run_auto_commit
    self.elastic.indices.refresh()
  File "/usr/lib/python2.6/site-packages/elasticsearch/client/utils.py", line 68, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/lib/python2.6/site-packages/elasticsearch/client/indices.py", line 53, in refresh
    params=params)
  File "/usr/lib/python2.6/site-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/usr/lib/python2.6/site-packages/elasticsearch/connection/http_urllib3.py", line 86, in perform_request
    self._raise_error(response.status, raw_data)
  File "/usr/lib/python2.6/site-packages/elasticsearch/connection/base.py", line 102, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.NotFoundError: TransportError(404, u'IndexMissingException[[*] missing]')

**Comment by lovett89** [comment]
@TissonMathew 
Does Elasticsearch have any indexes before starting Mongo Connector? It looks like this error is happening when Mongo Connector is trying to refresh all indexes on Elasticsearch, though I'm not really sure why an index would be "missing."

Can you please provide logs from Elasticsearch itself when this happens? Thanks.

**Comment by TissonMathew** [comment]
Please refer to #214

**Comment by TissonMathew** [comment]
Resolved.

**Comment by memayank00** [comment]
@TissonMathew  
when i typed  
mongo-connector -m localhost:27017 -t localhost:9200 --fields title,text -d mongo_connector/doc_managers/elastic_doc_manager.py

command to start  mongo-connector then i found error

No handlers could be found for logger "mongo_connector.util"
Traceback (most recent call last):
  File "/usr/local/bin/mongo-connector", line 9, in <module>
    load_entry_point('mongo-connector==2.1.dev0', 'console_scripts', 'mongo-connector')()
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector-2.1.dev0-py2.7.egg/mongo_connector/util.py", line 85, in wrapped
    func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector-2.1.dev0-py2.7.egg/mongo_connector/connector.py", line 999, in main
    conf.parse_args()
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector-2.1.dev0-py2.7.egg/mongo_connector/config.py", line 118, in parse_args
    option, dict((k, values.get(k)) for k in option.cli_names))
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector-2.1.dev0-py2.7.egg/mongo_connector/connector.py", line 779, in apply_doc_managers
    module = import_dm_by_name(dm['docManager'])
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector-2.1.dev0-py2.7.egg/mongo_connector/connector.py", line 769, in import_dm_by_name
    "Could not import %s." % full_name)
mongo_connector.errors.InvalidConfiguration: Could not import mongo_connector.doc_managers.mongo_connector/doc_managers/elastic_doc_manager.py

can you resolve this
.....

**Comment by llvtt** [comment]
I think your usage is wrong. Check out 
https://github.com/10gen-labs/mongo-connector/wiki/Usage%20with%20ElasticSearch. 
Also, please open a new issue for further help.

**Comment by memayank00** [comment]
@TissonMathew 
 yeah... it's working but after then this command suddenly mongo-connector became shutdown and into my log file  this is the error..

2015-05-06 14:00:37,838 [ERROR] mongo_connector.connector:277 - No replica set at "localhost:27017"! A replica set is required to run mongo-connector. Shutting down...
2015-05-06 15:06:32,370 [ERROR] mongo_connector.connector:277 - No replica set at "localhost:27017"! A replica set is required to run mongo-connector. Shutting down...
2015-05-06 15:06:44,661 [ERROR] mongo_connector.connector:277 - No replica set at "localhost:27017"! A replica set is required to run mongo-connector. Shutting down...
  

i know this is why because i can not create the replica set of mongodb
so can you please  suggest me how can i create replica set . actually i am brand new to mongodb so can you please suggest me ....
